dotCMS/src/main/java/com/dotmarketing/business/PermissionBitAPIImpl.java

```diff
@@ -1521,102 +1522,136 @@ public void permissionIndividuallyByRole(Permissionable parent, Permissionable p
-                    continue;
-                }
+            //Organize permission by role
+			final Map<String, List<Permission>> permissionsByRole = permissionList.stream()
+					.collect(Collectors.groupingBy(Permission::getRoleId));
 
-                if (type.equals(permission.getType()) || permission.isIndividualPermission()) {
-                    Permission duplicatedPermission = null;
-                    ImmutableList.Builder<Permission> immutableDuplicatedList = new Builder<>();
-
-                    for (Permission newPermission : newSetOfPermissions) {
-                        if (newPermission.isIndividualPermission() && newPermission.getRoleId()
-                                .equals(permission.getRoleId())
-                                && newPermission.getPermission() > permission
-                                .getPermission()) {
-                            duplicatedPermission = newPermission;
-                            break;
-                        } else if (newPermission.isIndividualPermission() && newPermission
-                                .getRoleId()
-                                .equals(permission.getRoleId())) {
-                            immutableDuplicatedList.add(newPermission);
-                        }
-                    }
-                    List<Permission> duplicatedPermissionList = immutableDuplicatedList.build();
-                    if (duplicatedPermission == null) {
-                        newSetOfPermissions.removeAll(duplicatedPermissionList);
-                        if (permission.isIndividualPermission()) {
-                            newSetOfPermissions.add(new Permission(permission.getType(),
-                                    permissionable.getPermissionId(), permission.getRoleId(),
-                                    permission.getPermission(), true));
-                            continue;
-                        } else {
-                            newSetOfPermissions.add(new Permission(permissionable.getPermissionId(),
-                                    permission.getRoleId(), permission.getPermission(), true));
-                        }
-                    }
-                    if (!permission.isIndividualPermission()) {
-                        newSetOfPermissions
-                                .add(new Permission(permission.getType(),
-                                        permissionable.getPermissionId(),
-                                        permission.getRoleId(), permission.getPermission(), true));
-                    }
-                } else {
-                    newSetOfPermissions
-                            .add(new Permission(permission.getType(),
-                                    permissionable.getPermissionId(),
-                                    permission.getRoleId(), permission.getPermission(), true));
-                }
-            }
+			final String finalPermissionableType = permissionableType;

```

Comment:
This `finalPermissionableType` is not needed, right? You can filter using the `permissionableType` as it doesn't change once assigned

Reply:
it's necessary since it is accessed within a forEach that requires only effective final variables to be used there